ambry-network/src/test/java/com.github.ambry.network/SSLSelectorTest.java

```diff
@@ -203,6 +202,36 @@ public void testEmptyRequest()
     assertEquals("", blockingRequest(connectionId, ""));
   }
 
+  @Test
+  public void testSSLConnect()
+      throws IOException {
+    String connectionId =
+        selector.connect(new InetSocketAddress("localhost", server.port), BUFFER_SIZE, BUFFER_SIZE, PortType.SSL);
+    Assert.assertFalse("Channel should not be ready by now (until handshake completes)",
+        selector.isChannelReady(connectionId));
+    while (!selector.connected().contains(connectionId)) {
+      selector.poll(10000L);
+    }
+    Assert.assertTrue("Channel should have been ready by now ", selector.isChannelReady(connectionId));
+  }
+
+  @Test
+  public void testCloseDuringSslHandshake()
+      throws IOException {
+    String connectionId =
+        selector.connect(new InetSocketAddress("localhost", server.port), BUFFER_SIZE, BUFFER_SIZE, PortType.SSL);
+    Assert.assertFalse("Channel should not be ready by now (until handshake completes)",
+        selector.isChannelReady(connectionId));
+    selector.poll(10000L);
+    Assert.assertFalse("Channel shouldn't have been added to connect list w/o handshake",
+        selector.connected().contains(connectionId));
+    selector.close(connectionId);
+    Assert.assertFalse("Channel should not have been added to connected list ",
+        selector.connected().contains(connectionId));
+    Assert.assertTrue("Channel should have been added to disconnected list",
+        selector.disconnected().contains(connectionId));
```

Comment:
Better also check `pendingHandshakeTransmissions` does not contain the `connectionId`.